packages/material-ui/src/styles/MuiThemeProvider.js

```diff
@@ -1,164 +1,7 @@
-    }
-  }
-
-  // Simple merge between the outer theme and the local theme
-  mergeOuterLocalTheme(localTheme) {
-    // To support composition of theme.
-    if (typeof localTheme === 'function') {
-      warning(
-        this.outerTheme,
-        [
-          'Material-UI: you are providing a theme function property ' +
-            'to the MuiThemeProvider component:',
-          '<MuiThemeProvider theme={outerTheme => outerTheme} />',
-          'However, no outer theme is present.',
-          'Make sure a theme is already injected higher in the React tree ' +
-            'or provide a theme object.',
-        ].join('\n'),
-      );
-      return localTheme(this.outerTheme);
-    }
-
-    if (!this.outerTheme) {
-      return localTheme;
-    }
-
-    return { ...this.outerTheme, ...localTheme };
-  }
-
-  render() {
-    // TODO move the sheetsManager property to a different component.
-    // warning(
-    //   typeof window !== 'undefined' || this.props.sheetsManager,
-    //   [
-    //     'Material-UI: you need to provide a sheetsManager to the MuiThemeProvider ' +
-    //       'when rendering on the server.',
-    //     'If you do not, you might experience a memory leak',
-    //   ].join('\n'),
-    // );
-    return this.props.children;
-  }
-}
-
-MuiThemeProviderOld.propTypes = {
-  /**
-   * You can wrap a node.
-   */
-  children: PropTypes.node.isRequired,
-  /**
-   * You can disable the generation of the styles with this option.
-   * It can be useful when traversing the React tree outside of the HTML
-   * rendering step on the server.
-   * Let's say you are using react-apollo to extract all
-   * the queries made by the interface server-side.
-   * You can significantly speed up the traversal with this property.
-   */
-  disableStylesGeneration: PropTypes.bool,
-  /**
-   * @ignore
-   *
-   * In beta.
-   */
-  sheetsCache: PropTypes.object,
-  /**
-   * The sheetsManager is used to deduplicate style sheet injection in the page.
-   * It's deduplicating using the (theme, styles) couple.
-   * On the server, you should provide a new instance for each request.
-   */
-  sheetsManager: PropTypes.object,
-  /**
-   * A theme object.
-   */
-  theme: PropTypes.oneOfType([PropTypes.object, PropTypes.func]).isRequired,
-};
-
-if (process.env.NODE_ENV !== 'production') {
-  MuiThemeProviderOld.propTypes = exactProp(MuiThemeProviderOld.propTypes);
-}
-
-MuiThemeProviderOld.childContextTypes = {
-  ...themeListener.contextTypes,
-  muiThemeProviderOptions: PropTypes.object,
-};
-
-MuiThemeProviderOld.contextTypes = {
-  ...themeListener.contextTypes,
-  muiThemeProviderOptions: PropTypes.object,
-};
-
-if (!ponyfillGlobal.__MUI_STYLES__) {
-  ponyfillGlobal.__MUI_STYLES__ = {};
-}
-
-if (!ponyfillGlobal.__MUI_STYLES__.MuiThemeProvider) {
-  ponyfillGlobal.__MUI_STYLES__.MuiThemeProvider = MuiThemeProviderOld;
-}
-
-export default ponyfillGlobal.__MUI_STYLES__.MuiThemeProvider;
+export default ThemeProvider;
```

Comment:
If it still exists we should include the types.

Reply:
It's only here until I figure out how to handle the CDN demo.